OpenRA.Mods.Common/Activities/Air/Fly.cs

```diff
@@ -37,6 +37,9 @@ public Fly(Actor self, Target t, WDist minRange, WDist maxRange)
 
 		public static void FlyToward(Actor self, Plane plane, int desiredFacing, WDist desiredAltitude)
 		{
+			var cell = self.World.Map.CellContaining(plane.CenterPosition);
+			desiredAltitude = desiredAltitude + new WDist(self.World.Map.MapHeight.Value.Contains(cell) ? 512 * self.World.Map.MapHeight.Value[cell] : 0);
```

Comment:
These need to use `map.DistanceAboveTerrain` instead of rolling their own checks.